src/main/java/com/nu/art/cyborg/modules/ScreenOptionsModule.java

```diff
@@ -26,21 +30,32 @@ protected void init() {
 
 	public final void setActivity(Activity activity) {
 		if (activity == null) {
-			window = null;
+			weakRefActivity = null;
 			return;
 		}
-		window = new WeakReference<Window>(activity.getWindow());
+		weakRefActivity = new WeakReference<Activity>(activity);
+	}
+
+	private Window getWindow() {
+		if (weakRefActivity == null || weakRefActivity.get() == null)
+			return null;
+
+		return weakRefActivity.get().getWindow();
+	}
+
+	private Activity getActivity() {
+		return weakRefActivity.get();
 	}
 
 	/**
 	 * brightnessLevel should be between 0 and 255
 	 */
 	public void setWindowBrightness(int brightnessLevel) {
-		if (this.window == null) {
+		if (getWindow() == null) {
```

Comment:
calling this method twice is not good you should hold a ref to the window as a local variable. the second call it might be null, null, null... to.. to.. tom..

Reply:
I mean if you also combine my upper comment